mava/systems/q_learning/types.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, NamedTuple, Tuple  # noqa
+
+import optax
+from chex import PRNGKey
+from flashbax.buffers.trajectory_buffer import TrajectoryBufferState
+from flax.core.scope import FrozenVariableDict
+from jax import Array
+from jumanji.env import State
+from typing_extensions import TypeAlias
+
+from mava.types import Observation
+
+Metrics = Dict[str, Array]
+
+
+class Transition(NamedTuple):
+    """Transition for recurrent Q-learning."""
+
+    obs: Observation
+    action: Array
+    reward: Array
+    terminal: Array
+    term_or_trunc: Array
+    # Even though we use a trajectory buffer we need to store both obs and next_obs.
+    # This is because of how the `AutoResetWrapper` returns obs at the end of an episode.
+    next_obs: Observation
+
+
+BufferState: TypeAlias = TrajectoryBufferState[Transition]
+
+
+class DDQNParams(NamedTuple):
```

Comment:
Since online and target params are used for other algorithms and not only for q learning systems can we name this something more generic like `QNetParams`

Reply:
Sounds good, done!